packages/fuels/tests/harness.rs

```diff
@@ -3596,3 +3597,36 @@ async fn testnet_hello_world() -> Result<(), Error> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_input_message() -> Result<(), Error> {
+    abigen!(
+        MyContract,
+        "packages/fuels/tests/test_projects/contract_test/out/debug/contract_test-abi.json"
+    );
+    let mut wallet = WalletUnlocked::new_random(None);
+
+    let messages = setup_single_message(wallet.address(), DEFAULT_COIN_AMOUNT);
+
+    let (provider, _) = setup_test_provider(vec![], None, Some(messages)).await;
+    wallet.set_provider(provider);
+
+    let contract_id = Contract::deploy(
+        "tests/test_projects/contract_test/out/debug/contract_test.bin",
+        &wallet,
+        TxParameters::new(None, None, None),
+        StorageConfiguration::default(),
+    )
+    .await?;
+
+    let contract_instance_connected =
+        MyContractBuilder::new(contract_id.to_string(), wallet.clone()).build();
+
+    let response = contract_instance_connected
+        .initialize_counter(42) // Build the ABI call
+        .call()
+        .await?;
+    assert_eq!(42, response.value);
+
+    Ok(())
```

Comment:
Could we add a check by calling `get_messages` or something similar to verify that the message we initialized the test client with exists?

Reply:
I added a function that partially compares Vec<Message>. It can also be improved if there is a need

Reply:
Which function?

Reply:
`compare_messages(messages_from_provider, messages) ` if you want to check if  `get_messages` is returning  same messages as the messages that we used to init. client. But maybe we don't need this?

Reply:
We should still check this variant here inside _this_ specific test, no?

Reply:
Oh, sorry I didn't see line 3636. I guess it could be useful to keep it in its own function if other tests want to also check it for some reason? Or would that be out of scope for other tests to check?

Reply:
I put it in the scope.